moveit_kinematics/ikfast_kinematics_plugin/templates/ikfast61_moveit_plugin_template.cpp

```diff
@@ -578,29 +578,29 @@ void IKFastKinematicsPlugin::getSolution(const IkSolutionList<IkReal>& solutions
 
 double IKFastKinematicsPlugin::harmonize(const std::vector<double>& ik_seed_state, std::vector<double>& solution) const
 {
-  double dist_sqr = 0;
+  double dist_abs = 0;
   std::vector<double> ss = ik_seed_state;
   for (size_t i = 0; i < ik_seed_state.size(); ++i)
   {
-    while (ss[i] > 2 * M_PI)
+    while (ss[i] > M_PI)
     {
       ss[i] -= 2 * M_PI;
     }
-    while (ss[i] < 2 * M_PI)
+    while (ss[i] < -1 * M_PI)
     {
       ss[i] += 2 * M_PI;
     }
-    while (solution[i] > 2 * M_PI)
+    while (solution[i] > M_PI)
     {
       solution[i] -= 2 * M_PI;
     }
-    while (solution[i] < 2 * M_PI)
+    while (solution[i] < -1 * M_PI)
     {
       solution[i] += 2 * M_PI;
     }
-    dist_sqr += fabs(ik_seed_state[i] - solution[i]);
+    dist_abs += fabs(ss[i] - solution[i]);
   }
-  return dist_sqr;
+  return dist_abs;
```

Comment:
I will add the comment explaining this. Actually, this function **harmonize** wasn't written correctly. I think the the solution output from solve(frame, vfree, solutions) as well as seed input can be something out of range (not between -pi to pi). So, this function transforms the solution in the range(-pi to pi) and finds its first norm from the transformed seed value. 
Originally, this function was not being used. But I found it useful in my implementation. So, I finally modified this because it was running on wrong definition.